runtime/compiler/aarch64/codegen/J9TreeEvaluator.cpp

```diff
@@ -39,3 +51,43 @@ J9::ARM64::TreeEvaluator::generateFillInDataBlockSequenceForUnresolvedField(TR::
    TR_ASSERT_FATAL(false, "This helper implements platform specific code for Fieldwatch, which is currently not supported on ARM64 platforms.\n");
    }
 	
+TR::Register *J9::ARM64::TreeEvaluator::DIVCHKEvaluator(TR::Node *node, TR::CodeGenerator *cg)
+   {
+   TR::Node *divisor = node->getFirstChild()->getSecondChild();
+   TR::Register *srcReg = NULL;
+   TR::Register *zeroReg = NULL;
+   bool is64Bit = node->getType().isInt64();
+   bool isConstDivisor = divisor->getOpCode().isLoadConst();
+   TR::LabelSymbol *snippetLabel = NULL;
+
+   if (!isConstDivisor || (!is64Bit && divisor->getInt() == 0) || (is64Bit && divisor->getLongInt() == 0))
+      {
+      snippetLabel = generateLabelSymbol(cg);
+      cg->addSnippet(new (cg->trHeapMemory()) TR::ARM64HelperCallSnippet(cg, node, snippetLabel, node->getSymbolReference()));
+
+      if (isConstDivisor)
+         {
+         // No explicit check required
+         generateLabelInstruction(cg, TR::InstOpCode::b, node, snippetLabel, (TR::Instruction*)NULL);
+         }
+      else
+         {
+         srcReg = cg->evaluate(divisor);
+         if (is64Bit)
+            {
+            loadConstant64(cg, node, 0, zeroReg, NULL);
+            }
+         else
+            {
+            loadConstant32(cg, node, 0, zeroReg, NULL);
+            }
+         generateCompareInstruction(cg, node, srcReg, zeroReg, is64Bit, NULL);
+         generateConditionalBranchInstruction(cg, TR::InstOpCode::b_cond, node, snippetLabel, TR::CC_EQ, NULL, NULL);
```

Comment:
Rather than loading a constant zero, comparing, and branching you should use the single `CBZ` instruction to branch to the snippet directly instead.